dali/base/dadfs.cpp

```diff
@@ -8866,18 +8867,21 @@ class CFileScanner
             CFileMatch *match = &files.item(f);
             matchingFiles.append(*LINK(match));
             ++count;
+            if (checkFileCount && (count > ITERATE_FILTEREDFILES_LIMIT))
+                throw MakeStringException(-1, "More than %d files found, please use a filter to reduce the number of files returned.",
+                    ITERATE_FILTEREDFILES_LIMIT);
```

Comment:
If the sorting was done at the server instead of the client, then yeah it could gather the top N without gathering all.
Beyond the scope of 5.2.4 though and this whole area probably needs a rethink for when we move files to a Cassandra like implementation.